### PR TITLE
add special collection field type

### PR DESCRIPTION
**Pull request by nicolasThal**

### CODE REVIEW

File: en/developer_guide/content_add_special_collection_field.rst [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+-------
+
+When working with the Content and ContentType, you might have the need to save a specific field such as a collection.
+
+In some case, this collection needs to be initialized by a listener to set all the datas.
+
+The main issues is that all the contentAttribute are stored as a raw field in mongoDB. By doing this, the data you will
+save will be typed but the data you get back will only be an array.
+
+I will present a exemple of how you can take profit of the Content architecture to store some TranslatedValue field
```
Comment (ngilain) [inline en/developer_guide/content_add_special_collection_field.rst]: Maybe something like "Here is an example of how..." to get you out of the doc

File: en/developer_guide/content_add_special_collection_field.rst [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+As you use a dynamic Field Type to describe your content, the ``Content`` document cannot implement the
+``TranslatedValueContainerInterface`` to give the name of the translated fields.
+
+Moreover, the listener have to be configured in the root form. As the configuration is fully dynamic, you cannot add it.
+
+Solution
+~~~~~~~~
+
+A simple solution would be to create a specific form type, add an initializer, and take care of the data transformation.
```
Comment (ngilain) [inline en/developer_guide/content_add_special_collection_field.rst]: a simple solution is to create

File: en/developer_guide/content_add_special_collection_field.rst [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                    $data['value'][$key] = $newElement;
+                }
+            }
+            $event->setData($data);
+        });
+
+        $builder->add('value', 'translated_value_collection');
+    }
+
+I will now explain all the different action!
```
Comment (ngilain) [inline en/developer_guide/content_add_special_collection_field.rst]: I think this sentence can be omitted

File: en/developer_guide/content_add_special_collection_field.rst [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+``ArrayCollection`` which will be later saved and use the initializer to generate all the data you need.
+
+This is the same workflow as the other translated values.
+
+As you can see, there is no event listener bind on Submit events. The raw field type automatically accept ``array`` and
+``ArrayCollection``.
+
+When you want to display the data stored, you need to transform them from an ``array`` to a ``TranslatedValue`` document.
+
+*This should be done in a DataTransformer.*
```
Comment (ngilain) [inline en/developer_guide/content_add_special_collection_field.rst]: Maybe a symfony link to the official datatransformer doc

File: en/developer_guide/content_add_special_collection_field.rst [hunk 5]
```diff
@@ -0,0 +1,132 @@
+Add a special collection field to a Content Attribute
+=====================================================
+
+Context
+-------
+
+When working with the Content and ContentType, you might have the need to save a specific field such as a collection.
```
Comment (ngilain) [inline en/developer_guide/content_add_special_collection_field.rst]: I think you should present the doc by talking of the listener issue before talking of the collection example.

**Comment by nicolasThal** [comment]
@itkg-ngilain update done

**Comment by nicolasThal** [comment]
@itkg-yfarci and @itkg-agilles, Is this documentation ok for you?

File: en/developer_guide/content_add_field_listener.rst [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+        $builder->add('value', 'translated_value_collection');
+    }
+
+When you create the contentAttribute, there are no data stored, so the ``$data`` is null. You need to create an
+``ArrayCollection`` which will be later saved and use the initializer to generate all the data you need.
+
+This is the same workflow as the other translated values.
+
+As you can see, there is no event listener bind on Submit events. The raw field type automatically accept ``array`` and
```
Comment (juchi) [inline en/developer_guide/content_add_field_listener.rst]: binded
accepts

File: en/developer_guide/content_add_field_listener.rst [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+``ArrayCollection``.
+
+When you want to display the data stored, you need to transform them from an ``array`` to a ``TranslatedValue`` document.
```
Comment (juchi) [inline en/developer_guide/content_add_field_listener.rst]: I think "data" is singular, so : them => it

File: en/developer_guide/content_add_field_listener.rst [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+
+The entity should also implement the ``TranslatedValueContainerInterface``.
+
+Issue description
+~~~~~~~~~~~~~~~~~
+
+As you use a dynamic Field Type to describe your content, the ``Content`` document cannot implement the
+``TranslatedValueContainerInterface`` to give the name of the translated fields.
+
+Moreover, the listener have to be configured in the root form. As the configuration is fully dynamic, you cannot add it.
```
Comment (juchi) [inline en/developer_guide/content_add_field_listener.rst]: has to be

**Comment by nicolasThal** [comment]
@juchi update done

**Comment by itkg-rboessel** [comment]
:+1:

File: en/developer_guide/content_add_field_listener.rst [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+.. code-block:: php
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $translatedValueDefaultValueInitializer = $this->translatedValueDefaultValueInitializer;
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, function(FormEvent $event) use ($translatedValueDefaultValueInitializer) {
+            $data = $event->getData();
+            if (is_null($data)) {
+                $values = new ArrayCollection();
+                $translatedValueDefaultValueInitializer->generate($values);
```
Comment (itkg-yfarci) [inline en/developer_guide/content_add_field_listener.rst:92]: J'ai une erreur à cet endroit " Attempted to call method "generate" on class "OpenOrchestra\BackofficeBundle\EventListener\TranslateValueInitializerListener". "
Effectivement la fonction generate n'existe pas dans la class "TranslateValueInitializerListener"

Comment (nicolasThal) [inline en/developer_guide/content_add_field_listener.rst:92]: Sorry, this documentation is prepared for the 0.2.3 version. In your current version, the class TranslatedValueDefaultValueInitializer does not exist.

**Comment by ngilain** [comment]
@itkg-yfarci , @itkg-gotowy : is that doc ok for you ? If yes, we can merge it in the master, if no, let us know what is lacking please

**Comment by itkg-yfarci** [comment]
yes it's good for us :+1:

**Pull request merged**